### PR TITLE
Devs

### PR DESCRIPTION
**Pull request by bryanm92s**
Se realizan correciones 

**1.** Se eliminan líneas innecesarias.
**2.** Se implementa Binding en Fragments, Adapters, RecyclerView
**3.** Se implementa Strings, Dimens.
**4.** Se crea fragmento Movie Details.
**5.** Se realizan adecuaciones al diseño de la aplicación.

Inicio
Buscar
Perfil
MovieDetails

### CODE REVIEW

**Comment by bryanm92s** [comment]
@jrojas25

File: app/src/main/java/com/example/apppractica/Views/SignUp.kt [hunk 1]
```diff
@@ -20,21 +20,24 @@ class SignUp : AppCompatActivity() {
...
+        setContentView(binding.root)
 
         //calling view model object
         val userDetailsRepository = ViewModelProvider(this@SignUp).get(LoginViewModel::class.java)
 
 
-        btn_back_login.setOnClickListener {
+        binding.btnBackLogin.setOnClickListener {
             val intent: Intent = Intent(this@SignUp, Login::class.java)
             startActivity(intent)
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/Views/SignUp.kt:37]: is always a good practice to finish the current activity when we don't want it to remain in the stack (if we don't do this when the user taps the back button will be redirected to this screen again)

File: app/src/main/java/com/example/apppractica/Views/SignUp.kt [hunk 2]
```diff
@@ -43,7 +46,7 @@ class SignUp : AppCompatActivity() {
 
                         for (i in userObject.indices) {
 
-                            if (userObject[i].name?.equals(textInputEditTextNombre.text.toString())!!) {
+                            if (userObject[i].name?.equals(binding.textInputEditTextNombre.text.toString())!!) {
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/Views/SignUp.kt:49]: avoid this validation, and the whole logic of going to the database and get all the users, instead let the database return an error when the user already exists

File: app/src/main/java/com/example/apppractica/Views/SignUp.kt [hunk 3]
```diff
@@ -74,41 +77,40 @@ class SignUp : AppCompatActivity() {
                                 .show()
 
                             // Limpiar EditText
-                            textInputEditTextNombre.setText("")
-                            textInputEditTextCorreo.setText("")
-                            textInputEditTextContrasena.setText("")
+                            binding.textInputEditTextNombre.setText("")
+                            binding.textInputEditTextCorreo.setText("")
+                            binding.textInputEditTextContrasena.setText("")
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/Views/SignUp.kt:82]: move this to its own method, remember the Single Responsibility principle from SOLID

File: app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt [hunk 4]
```diff
@@ -2,70 +2,55 @@
...
 
 class BestMoviesAdapter(private val bestMovieList: List<BestMovies>) : androidx.recyclerview.widget.RecyclerView.Adapter<BestMoviesAdapter.ViewHolder>() {
 
 
-  override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+  /*override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
     val inflater = LayoutInflater.from(viewGroup.context)
     val view = inflater.inflate(R.layout.item_horizontal, viewGroup, false)
     return ViewHolder(view)
+  }*/
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/adapter/BestMoviesAdapter.kt:19]: remove the unused code

File: app/src/main/res/layout/activity_login.xml [hunk 5]
```diff
@@ -35,7 +35,7 @@
                 android:layout_height="wrap_content"
                 android:layout_marginStart="32dp"
                 android:layout_marginTop="16dp"
-                android:text="Usuario"
+                android:text="@string/usuario"
                 android:textColor="@color/color_gris_oscuro"
                 android:textSize="20sp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml:40]: use dimens

File: app/src/main/res/layout/activity_login.xml [hunk 6]
```diff
@@ -35,7 +35,7 @@
                 android:layout_height="wrap_content"
                 android:layout_marginStart="32dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_login.xml:36]: use dimens

File: app/src/main/res/layout/activity_login.xml (hunk 5 above)
Comment (jrojas25) [reply app/src/main/res/layout/activity_login.xml:40]: and check the all the other layouts to fix the same

File: app/src/main/res/layout/fragment_inicio.xml [hunk 7]
```diff
@@ -13,65 +15,113 @@
...
+                            <TextView
+                                android:layout_width="wrap_content"
+                                android:layout_height="wrap_content"
+                                android:layout_marginStart="20dp"
+                                android:text="@string/trailer_oficial"
+                                android:textSize="14sp">
+                            </TextView>
+                        </LinearLayout>
+                    </LinearLayout>
+                </RelativeLayout>
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_inicio.xml:75]: avoid nesting layouts and use a constraint layout instead

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+        android:id="@+id/tv_back"
+        android:layout_width="52dp"
+        android:layout_height="35dp"
+        android:layout_marginTop="25dp"
+        android:background="@drawable/btn_icon_back"
+        app:layout_constraintEnd_toStartOf="@id/titulo_ppal"
+        app:layout_constraintHorizontal_bias="0.20"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="SpeakableTextPresentCheck,SpeakableTextPresentCheck,TouchTargetSizeCheck,TouchTargetSizeCheck" />
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_movie_details.xml:19]: never use the `tools:ignore`

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/divisor_uno"
+        android:orientation="vertical"
+        android:layout_marginHorizontal="45dp"
+        >
+
+        <LinearLayout
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_movie_details.xml:62]: you're already suing a constraint layout, so there's no need to nest or even use Linear layouts here

File: app/src/main/res/layout/fragment_perfil.xml [hunk 10]
```diff
@@ -1,15 +1,284 @@
...
+                android:orientation="vertical"
+                android:layout_marginTop="10dp"
+                app:layout_constraintTop_toBottomOf="@id/card2"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                android:layout_marginBottom="20dp"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+
+                <LinearLayout
```
Comment (jrojas25) [inline app/src/main/res/layout/fragment_perfil.xml:114]: don't nest layouts, and don't use other layouts when you're already using constraint layouts

**Review by jrojas25 — CHANGES_REQUESTED** [review]

**Pull request merged**